utils/python/CIME/XML/env_mach_pes.py

```diff
@@ -14,3 +14,29 @@ def __init__(self, case_root=os.getcwd(), infile="env_mach_pes.xml"):
         initialize an object interface to file env_mach_pes.xml in the case directory
         """
         EnvBase.__init__(self, case_root, infile)
+
+    def get_value(self, vid, attribute={}, resolved=True, subgroup=None):
+        value = EnvBase.get_value(self, vid, attribute, resolved, subgroup)
+        if "NTASKS" in vid and value < 0:
+            value = -1*value*self.get_value("PES_PER_NODE")
+        if "NTHRDS" in vid and value < 0:
+            value = -1*value*self.get_value("PES_PER_NODE")
+        if "ROOTPE" in vid and value < 0:
+            value = -1*value*self.get_value("PES_PER_NODE")
+        return value
+#
+# We need a set value until we full transition from perl
+#
+
+    def _set_value(self, node, value, vid=None, subgroup=None, ignore_type=False):
+        if vid is None:
+            vid = node.get("id")
+
```

Comment:
But the perl code can't handle it so we need this correction until we eliminate the perl